### DISCUSSION TITLE
IWP 에서 구매 후 ....

### DISCUSSION THREAD

**Discussion by peterkangdev** [naverz/zepeto-studio-kor#405] [5. Q&A]
onPurchased() 함수가 구매 할때만 연동되는 것 같은데, 
구매가 완료된 후 나중에 접속했을때 구매 정보는 어디서 얻을 수 있나요?

**Comment by jooseungyou** [comment]
IWP에서 현재 별도로 이전 history의 receipt를 조회할 수 있는 기능은 제공되지 않고 있습니다...!

구매정보를 직접 storing해서 나중에 접속했을때 불러오셔야하는데, 해당 문서들은 참고하셔서 서버에 구매 이후 발급되는 receipt나 custom한 정보를 storing 하시면 될 것 같습니다.

(receipt): https://studio.zepeto.me/kr/guides/handling-iwp-event(server)
(storing): https://studio.zepeto.me/kr/guides/multiplay-game-datastorage

**Reply by peterkangdev** [reply]
만약 구매 버튼을 누른 후 어떠한 이유로 서버와 연결이 끊기는 상황이 오고 저장하지 못했다면,
구매를 완료한 후 구매 정보를 저장하지 못한 상태에서 앱이 죽거나 중단된다면 ... (생각하기도 싫은데...) 

이런 상황이 오지 말란법은 없습니다. 

iOS 같은 경우 Restore 기능이 있습니다. 

대책이 따로 있으신지.. 궁금합니다.

**Reply by jooseungyou** [reply]
안녕하세요 peterkangdev님!
말씀 주신 구매 history를 읽어 올 수 있는 함수를 다음 빌드 버젼에서 제공할 예정입니다!

다음 빌드부터 ZepetoIWPService.GetReceipts({userId}, {worldId}, ...)를 사용하시면 해당 유저가 해당 월드에서 구매한 이력들이 다음과 같은 형태로 제공 될 예정입니다.

{receiptId: ZPT-5d0a7e7e-bdcb7-b08aa, itemId: ah.eh, worldId: com.testing.iwp, purchasePrice: 2, status: purchased, purchasedAt: 2021-10-25T00:24:05.172Z, createdAt: 2021-10-25T00:24:04.955Z, updatedAt: 2021-10-25T00:24:05.483Z, IsPurchased: True, PurchasedAt: 2021-10-25 AM 9:24:05, CreatedAt: 2021-10-25 AM 9:24:04, UpdatedAt: 2021-10-25 AM 9:24:05, ReceiptStatus: Purchased}

항상 제페토의 발전에 도움주셔서 감사합니다!

**Reply by peterkangdev** [reply]
너무 감사합니다. 
항상 좋은 의견이 있으면 건의 드리겠습니다.
